.github/workflows/publish-base.yml

```diff
@@ -0,0 +1,106 @@
+    secrets:
+      # The target path within the host share to save files to
+      target-path:
+        required: true
+        description: |
+          The path within the host machine where the data should be published
+          to.
+      ssh-key:
+        required: true
+        description: The ssh private key that is used to publish to the host.
+      host:
+        required: true
+        description: The host that has sshd running to publish to.
+      port:
+        required: true
+        description: The port the host is running sshd on.
+      user:
+        required: true
+        description: THe username to log in to the server as.
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      # If this is somehow not run from a tag, fail. This is a remainder from
+      # when the build side and publish side were done in one workflow.
+      - name: Check tag status
+        if: |
+          startsWith(github.ref, 'refs/tags/') 
+            && ! endsWith(github.ref, inputs.tag)
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed("Provided tag does not match github ref tag")
+      # Download the requested github artifact
+      - name: Download artifacts
+        id: download
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.github-artifact-name }}
+      # Get the path to where the artifact is at. This is the path to the
+      # downloaded artifact that is relative to the github workspace with the
+      # source-path parameter tacked on the end.
+      - name: Get Source Path
+        shell: bash
+        id: get-source-path
+        run: |
+          echo "path=$( \
+            realpath -s --relative-to="${{ github.workspace }}" \
+            "${{ steps.download.outputs.download-path }}/${{ inputs.source-path }}" \
+            )" >> $GITHUB_OUTPUT
+      # Publish to the host
+      - name: Publish Release
+        uses: easingthemes/ssh-deploy@main
+        with:
+          SSH_PRIVATE_KEY: ${{ secrets.ssh-key }}
+          ARGS: "-vzrli"
+          SOURCE: ${{ steps.get-source-path.outputs.path }}
+          REMOTE_HOST: ${{ secrets.host }}
+          REMOTE_USER: ${{ secrets.user }}
+          REMOTE_PORT: ${{ secrets.port }}
+          TARGET: ${{ secrets.target-path }}/${{ inputs.target-artifact-path }}
+      # Create the release draft on GitHub.
+      - name: Create Release Draft
+        if: ${{ startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, inputs.tag) }}
+        uses: ncipollo/release-action@v1
+        with:
+          # artifacts: "artifacts-${{ github.sha }}/*"
+          # artifactErrorsFailBuild: true
+          draft: true
+          generateReleaseNotes: true
+          tag: ${{ inputs.tag }}
+          prerelease: ${{ contains(inputs.tag, '-beta') || contains(inputs.tag, '-alpha') || contains(inputs.tag, '-rc') }}
```

Comment:
any "suffix" of a semver tag is a prerelease (which we can also use for nightlies). If tag would be `1.2.3-dev.12345+abcdef` it would not match, but we still want a publish/deploy action here, as nightlies are important as well

Reply:
Updated to pull the prerelease from the semver check and then mark as a prerelease based on that.